engine/src/main/java/org/destinationsol/game/projectile/ProjectileConfigs.java

```diff
@@ -38,40 +41,46 @@
-            OggSound workSound = workSoundUrn.isEmpty() ? null : soundManager.getSound(workSoundUrn);
-            boolean bodyless = node.getBoolean("massless", false);
-            float density = node.getFloat("density", -1);
-            float dmg = node.getFloat("dmg");
-            float emTime = node.getFloat("emTime", 0);
-            ProjectileConfig c = new ProjectileConfig(tex, texSz, spdLen, stretch, physSize, dmgType, collisionSound,
-                    lightSz, trailEffect, bodyEffect, collisionEffect, collisionEffectBg, zeroAbsSpd, origin, acc, workSound, bodyless, density, guideRotSpd, dmg, emTime);
-            myConfigs.put(node.name, c);
-        }
+        for (ResourceUrn configUrn : projectileConfigurationFiles) {
+            Json json = Assets.getJson(configUrn.toString());
+            JsonValue rootNode = json.getJsonValue();
+
+            for (JsonValue node : rootNode) {
+                String texName = node.getString("tex");
+                TextureAtlas.AtlasRegion tex = Assets.getAtlasRegion(configUrn.getModuleName() + ":" + texName + "Projectile");
+                float texSz = node.getFloat("texSz");
+                float spdLen = node.getFloat("spdLen");
+                float physSize = node.getFloat("physSize", 0);
+                boolean stretch = node.getBoolean("stretch", false);
+                DmgType dmgType = DmgType.forName(node.getString("dmgType"));
+                String collisionSoundUrn = node.getString("collisionSound", "");
+                OggSound collisionSound = collisionSoundUrn.isEmpty() ? null : soundManager.getSound(collisionSoundUrn);
+                float lightSz = node.getFloat("lightSz", 0);
+                EffectConfig trailEffect = EffectConfig.load(node.get("trailEffect"), effectTypes, cols);
+                EffectConfig bodyEffect = EffectConfig.load(node.get("bodyEffect"), effectTypes, cols);
+                EffectConfig collisionEffect = EffectConfig.load(node.get("collisionEffect"), effectTypes, cols);
+                EffectConfig collisionEffectBg = EffectConfig.load(node.get("collisionEffectBg"), effectTypes, cols);
+                float guideRotSpd = node.getFloat("guideRotSpd", 0);
+                boolean zeroAbsSpd = node.getBoolean("zeroAbsSpd", false);
+                Vector2 origin = SolMath.readV2(node.getString("texOrig", "0 0"));
+                float acc = node.getFloat("acceleration", 0);
+                String workSoundUrn = node.getString("workSound", "");
+                OggSound workSound = workSoundUrn.isEmpty() ? null : soundManager.getSound(workSoundUrn);
+                boolean bodyless = node.getBoolean("massless", false);
+                float density = node.getFloat("density", -1);
+                float dmg = node.getFloat("dmg");
+                float emTime = node.getFloat("emTime", 0);
+                ProjectileConfig c = new ProjectileConfig(tex, texSz, spdLen, stretch, physSize, dmgType,
```

Comment:
Can we rename c to something more.. meaningful? `projectileConfig` would be ideal.